File: services/education/internal/model/ledger.go
package model

import "time"

const (
	LedgerTypeIncome      = 1
	LedgerTypeExpenditure = 2
)

const (
	LedgerSourceSystem = 1
	LedgerSourceManual = 2
)

const (
	LedgerSystemTypeOrderPayment = 101
)

const (
	LedgerConfirmStatusPending      = 0
	LedgerConfirmStatusConfirmed    = 1
	LedgerConfirmStatusRefunding    = 2
	LedgerConfirmStatusRefundFailed = 3
)

const (
	LedgerCategoryOrderIncome      = "order-income"
	LedgerSubCategoryRegistration  = "registration-renewal"
	LedgerSubCategoryRefundCourse  = "refund-course"
	LedgerSubCategoryTransferOrder = "transfer-course"
)

type LedgerListQueryDTO struct {
	PageRequestModel PageRequestModel  `json:"pageRequestModel"`
	QueryModel       LedgerQueryFilter `json:"queryModel"`
	SortModel        map[string]any    `json:"sortModel"`
}

type LedgerQueryFilter struct {
	AccountIDs            []string `json:"accountIds"`
	LedgerConfirmStatuses []int    `json:"ledgerConfirmStatuses"`
	SourceTypes           []int    `json:"sourceTypes"`
	DealStaffID           string   `json:"dealStaffId"`
	ConfirmStaffID        string   `json:"confirmStaffId"`
	StudentID             string   `json:"studentId"`
	OrderNumber           string   `json:"orderNumber"`
	BankSlipNo            string   `json:"bankSlipNo"`
	LedgerNumber          string   `json:"ledgerNumber"`
	ConfirmStartTime      string   `json:"confirmStartTime"`
	ConfirmEndTime        string   `json:"confirmEndTime"`
	PayStartTime          string   `json:"payStartTime"`
	PayEndTime            string   `json:"payEndTime"`
	LedgerSubCategoryIDs  []string `json:"ledgerSubCategoryIds"`
	OrderID               string   `json:"orderId"`
}

type LedgerRichText struct {
	Text   string   `json:"text"`
	Images []string `json:"images"`
}

type LedgerListItemVO struct {
	ID                    string         `json:"id"`
	OrgID                 string         `json:"orgId,omitempty"`
	SchoolID              string         `json:"schoolId,omitempty"`
	Type                  int            `json:"type"`
	SourceType            int            `json:"sourceType"`
	LedgerCategoryID      string         `json:"ledgerCategoryId"`
	LedgerCategoryName    string         `json:"ledgerCategoryName"`
	LedgerSubCategoryID   string         `json:"ledgerSubCategoryId"`
	LedgerSubCategoryName string         `json:"ledgerSubCategoryName"`
	LedgerCategoryIcon    string         `json:"ledgerCategoryIcon"`
	Amount                float64        `json:"amount"`
	DealStaffID           string         `json:"dealStaffId,omitempty"`
	DealStaffName         string         `json:"dealStaffName,omitempty"`
	PayTime               *time.Time     `json:"payTime,omitempty"`
	CreatedTime           *time.Time     `json:"createdTime,omitempty"`
	PayMethod             *int           `json:"payMethod,omitempty"`
	AccountID             string         `json:"accountId,omitempty"`
	AccountName           string         `json:"accountName,omitempty"`
	ReciprocalAccount     string         `json:"reciprocalAccount"`
	BankSlipNo            string         `json:"bankSlipNo"`
	OrderNumber           string         `json:"orderNumber,omitempty"`
	LedgerNumber          string         `json:"ledgerNumber"`
	StudentID             string         `json:"studentId,omitempty"`
	StudentName           string         `json:"studentName,omitempty"`
	StudentPhone          string         `json:"studentPhone,omitempty"`
	IsConfirmed           bool           `json:"isConfirmed"`
	ConfirmRemark         LedgerRichText `json:"confirmRemark"`
	ProductItems          []string       `json:"productItems"`
	ConfirmStaffName      string         `json:"confirmStaffName"`
	ConfirmTime           *time.Time     `json:"confirmTime,omitempty"`
	SystemType            int            `json:"systemType"`
	OrderID               string         `json:"orderId,omitempty"`
	PaymentVoucher        LedgerRichText `json:"paymentVoucher"`
	BillFlowID            string         `json:"billFlowId,omitempty"`
	BillID                string         `json:"billId,omitempty"`
	LedgerConfirmStatus   int            `json:"ledgerConfirmStatus"`
	ErrorMessage          string         `json:"errorMessage"`
}

type LedgerListResultVO struct {
	List  []LedgerListItemVO `json:"list"`
	Total int                `json:"total"`
}

type LedgerStatisticsVO struct {
	IncomeAmount      float64 `json:"incomeAmount"`
	ExpenditureAmount float64 `json:"expenditureAmount"`
	BalanceAmount     float64 `json:"balanceAmount"`
	TotalConfirm      int     `json:"totalConfirm"`
	TotalUnConfirm    int     `json:"totalUnConfirm"`
	TotalRefunding    int     `json:"totalRefunding"`
	TotalRefundFailed int     `json:"totalRefundFailed"`
}

type LedgerOperateDTO struct {
	ID string `json:"id"`
}

File: 5.0-admin/src/api/finance-center/ledger.ts
import { usePost } from '~/utils/request'

export interface LedgerRichText {
  text?: string
  images?: string[]
}

export interface LedgerItem {
  id: string
  type: number
  sourceType: number
  ledgerCategoryId: string
  ledgerCategoryName: string
  ledgerSubCategoryId: string
  ledgerSubCategoryName: string
  ledgerCategoryIcon: string
  amount: number
  dealStaffId?: string
  dealStaffName?: string
  payTime?: string
  createdTime?: string
  payMethod?: number
  accountId?: string
  accountName?: string
  reciprocalAccount?: string
  bankSlipNo?: string
  orderNumber?: string
  ledgerNumber: string
  studentId?: string
  studentName?: string
  studentPhone?: string
  isConfirmed?: boolean
  confirmRemark?: LedgerRichText
  productItems?: string[]
  confirmStaffName?: string
  confirmTime?: string
  systemType?: number
  orderId?: string
  /** 关联订单类型：2 储值充值、4 储值退费（账单列表用于展示办理内容等） */
  orderType?: number
  /**
   * 储值账单办理内容用：储值账户名称（勿与 studentName、收款 accountName 混淆）。
   * 若列表接口未返回本字段，前端只显示「-」，需后端在 /api/v1/ledgers/list 结果中补齐。
   */
  rechargeAccountName?: string
  /** 部分后端可能使用的同义字段 */
  storedValueAccountName?: string
  paymentVoucher?: LedgerRichText
  billFlowId?: string
  billId?: string
  ledgerConfirmStatus: number
  errorMessage?: string
}

export interface LedgerListResult {
  list?: LedgerItem[]
  total?: number
}

export interface LedgerStatistics {
  incomeAmount?: number
  expenditureAmount?: number
  balanceAmount?: number
  totalConfirm?: number
  totalUnConfirm?: number
  totalRefunding?: number
  totalRefundFailed?: number
}

export interface LedgerQueryParams {
  sortModel?: Record<string, any>
  queryModel?: {
    accountIds?: string[]
    ledgerConfirmStatuses?: number[]
    sourceTypes?: number[]
    dealStaffId?: string
    confirmStaffId?: string
    studentId?: string
    orderNumber?: string
    bankSlipNo?: string
    ledgerNumber?: string
    confirmStartTime?: string
    confirmEndTime?: string
    payStartTime?: string
    payEndTime?: string
    ledgerSubCategoryIds?: string[]
    orderId?: string
  }
  pageRequestModel: {
    needTotal?: boolean
    pageSize: number
    pageIndex: number
    skipCount?: number
  }
}

export function getLedgerListApi(data: LedgerQueryParams) {
  return usePost<LedgerListResult>('/api/v1/ledgers/list', data)
}

export function getLedgerStatisticsApi(data: LedgerQueryParams) {
  return usePost<LedgerStatistics>('/api/v1/ledgers/statistics', data)
}

export function confirmLedgerApi(data: { id: string, confirmRemark?: LedgerRichText }) {
  return usePost<void>('/api/v1/ledgers/confirm', data)
}

export function cancelConfirmLedgerApi(data: { id: string }) {
  return usePost<void>('/api/v1/ledgers/cancel-confirm', data)
}
